### PR TITLE
Dropped outdated arcfour cipher

### PR DESCRIPTION
**Pull request by dasm**
OpenSSL 7.6 removed support for arcfour. rsync which uses ssh is trying
to establish connection with use of arcfour. For newer installations it
is ending with an error to establish connection.

Removed hardcoded arcfour cipher from code.

---
This change is [Reviewable](https://reviewable.io/reviews/pytest-dev/pytest-cloud/12)

### CODE REVIEW

**Comment by dasm** [comment]
Identified the issue with outdated `arcfour` cipher. https://github.com/pytest-dev/pytest-cloud/issues/11

File: pytest_cloud/rsync.py [hunk 1]
```diff
@@ -86,7 +86,7 @@ def send(self, raises=True):
                     '--inplace '
                     '--delete-excluded '
                     '--delete '
-                    '-e \"ssh -T -c arcfour -o Compression=no -x\" '
```
Comment (bubenkoff) [inline pytest_cloud/rsync.py:89]: as I remember, this cipher was passed to have the least possible network overhead, so what is the next another cipher we can pass from the supported ones?

**Review by bubenkoff — CHANGES_REQUESTED** [review]

File: pytest_cloud/rsync.py (hunk 1 above)
Comment (dasm) [reply pytest_cloud/rsync.py:89]: OK, that makes sense. Let me look around to see which cipher should replace `arcfour`.

Comment (dasm) [reply pytest_cloud/rsync.py:89]: Based on this old post: https://blog.famzah.net/2010/06/11/openssh-ciphers-performance-benchmark/ indeed the best was `arcfour`. But later, author updated his benchmark https://blog.famzah.net/2015/06/26/openssh-ciphers-performance-benchmark-update-2015/ and found that the fastest is `aes128-gcm@openssh.com`.

According to analysis, everything depends on type of platform. AES instruction set, introduced by Intel in 2010 and AMD in mid-2013 is significantly improving speeds of operations.

Do you think we should force ssh  to use this cipher? Would it make sense to leave decision about cipher up to ssh?

Comment (bubenkoff) [reply pytest_cloud/rsync.py:89]: let's use aes128-gcm@openssh.com then
the point of pytest-cloud that it's an opinionated approach, so the less to configure for the user is better

Comment (dasm) [reply pytest_cloud/rsync.py:89]: Replaced with `aes128-gcm@openssh.com`. Thanks

Comment (spinus) [reply pytest_cloud/rsync.py:89]: can't this option be just confuguration with sane default?

Comment (bubenkoff) [reply pytest_cloud/rsync.py:89]: makes sense

Comment (dasm) [reply pytest_cloud/rsync.py:89]: OK, I'll try making it config option.

Comment (dasm) [reply pytest_cloud/rsync.py:89]: @bubenkoff @spinus is it looking better now?

**Review by bubenkoff — CHANGES_REQUESTED** [review]
please add the docs about new option, also please add a changelog entry

File: README.rst [hunk 2]
```diff
@@ -85,6 +85,8 @@ Command-line options
     Optional process count limit for `rsync` processes. By default there's no limit so rsyncing will be in parallel
     for all test nodes.
 
+* `--cloud-rsync-cipher`
+    Optional ssh cipher selection for `rsync` processes. aes128-gcm@openssh.com by default.
```
Comment (bubenkoff) [inline README.rst:89]: please add a reason for choosing the non-default cipher

**Review by bubenkoff — CHANGES_REQUESTED** [review]

File: README.rst (hunk 2 above)
Comment (dasm) [reply README.rst:89]: Added more verbose description.

**Review by bubenkoff — APPROVED** [review]

**Comment by bubenkoff** [comment]
@dasm what is your pypi username, I will add you as a maintainer so that you can make releases yourself

**Review by bubenkoff — CHANGES_REQUESTED** [review]
@dasm sorry, one more thing, please also add yourself to contributors

**Comment by dasm** [comment]
@bubenkoff my pypi is the same (just created :P). Added myself to AUTHORS too.

**Review by bubenkoff — APPROVED** [review]

**Comment by bubenkoff** [comment]
@dasm please don't forget to make and push git tags when making the releases

**Comment by bubenkoff** [comment]
and of course, the version has to be changed

**Comment by dasm** [comment]
OK. Will change version and push tags. Thanks @bubenkoff

**Comment by dasm** [comment]
@bubenkoff I tried to publish new tag on repository, but it's giving me: `failed to push some refs`
I believe I don't have permissions to do so. For PyPI, `Manage` button is greyed out too.

Am I missing something?

**Comment by bubenkoff** [comment]
what's the git command you're running?
as for pypi, you don't have to manage it via the web interface, just do: 

```
 python setup.py sdist upload
```

from the project folder

**Comment by dasm** [comment]
I successfully uploaded new version to pypi with CLI. I also pushed new tag. Forgot to create tag with `git tag 4.0.0 master` at first place. Doh.

Thank you @bubenkoff

**Pull request merged**